src/Main.elm

```diff
@@ -1,10 +1,571 @@
+allNodesInDict =
+    let
+        getId : Node -> Id
+        getId node =
+            case node of
+                Dir id audienceFolder ->
+                    id
+
+                File id audience ->
+                    id
+
+                Root ->
+                    0
+    in
+    allNodes
+        |> List.map (\node -> ( getId node, node ))
+        |> Dict.fromList
+
+
+allNodes : List Node
+allNodes =
+    let
+        audienceFolders : List AudienceFolder
+        audienceFolders =
+            Maybe.withDefault [ defaultAudienceFolder ] decodeAudienceFolders
+
+        audiences : List Audience
+        audiences =
+            Maybe.withDefault [ defaultAudience ] decodeAudiences
+    in
+    List.append
+        (List.map (\audienceFolder -> Dir audienceFolder.id audienceFolder) audienceFolders)
+        (List.map (\audience -> File audience.id audience) audiences)
+
+
+
+-- Utility functions for traversing a Tree
+-- Finding a Tree with node
+-- Get children Nodes of certain parent tree
+
+
+findATree : Node -> Tree Node -> Maybe (Zipper.Zipper Node)
+findATree node tree =
+    Zipper.fromTree tree
+        |> Zipper.findFromRoot (\id -> id == node)
+
+
+getChildrenNodes : Tree Node -> List Node
+getChildrenNodes tree =
+    tree
+        |> Tree.children
+        |> List.map Tree.label
+
+
+
+-- UPDATE
+
+
+type Msg
+    = NoOp
+    | GoTo Node
+    | GoUp
+
+
+update : Msg -> Model -> ( Model, Cmd Msg )
+update msg model =
+    case msg of
+        NoOp ->
+            ( model, Cmd.none )
+
+        GoTo node ->
+            let
+                newZipperStep : Maybe (Zipper.Zipper Node)
+                newZipperStep =
+                    findATree node model.directory
+
+                updatedTree : Maybe (Tree Node)
+                updatedTree =
+                    newZipperStep
+                        |> Maybe.map Zipper.tree
+            in
+            ( { model | zipperStep = newZipperStep, currentTree = updatedTree }, Cmd.none )
+
+        GoUp ->
+            let
+                newZipperStep : Maybe (Zipper.Zipper Node)
+                newZipperStep =
+                    model.zipperStep
+                        |> Maybe.andThen Zipper.parent
+
+                updatedTree : Maybe (Tree Node)
+                updatedTree =
+                    newZipperStep
+                        |> Maybe.map Zipper.tree
+            in
+            ( { model | zipperStep = newZipperStep, currentTree = updatedTree }, Cmd.none )
+
+
+
+-- VIEW
+
+
+type alias Model =
+    { directory : Tree Node
+    , currentTree : Maybe (Tree Node)
+    , zipperStep : Maybe (Zipper.Zipper Node)
+    }
+
+
+view : Model -> Html Msg
+view model =
+    let
+        nodes : List Node
+        nodes =
+            case model.currentTree of
+                Just tree ->
+                    getChildrenNodes tree
+
+                Nothing ->
+                    []
+    in
+    div [ class "w-screen h-screen bg-gray-50" ]
```

Comment:
Just a random note: I was very impressed by the UI! Nice job!

Reply:
Thanks :)